### PR TITLE
Server refactor

### PR DESCRIPTION
**Pull request by googol**

### CODE REVIEW

**Comment by googol** [comment]
@supertassu If you'd like you can give this one a look over. It will get some more conflicts when we merge your second PR, but I'll handle those conflicts soon

File: test/functional/text-search.test.js [hunk 1]
```diff
@@ -103,11 +103,17 @@ describe('Free-text search in participant list API endpoint', () => {
     return expect(firstNames).to.have.members(expectedResult)
   }
 
-  async function queryParticipants(filter) {
+  async function queryParticipants({ textSearch, ...filter }) {
```
[the remark this answers is not in the document]
Comment (googol) [reply test/functional/text-search.test.js:106]: The conditional is actually there for a different reason. For the `URLSearchParams` constructor it makes a difference if a property is not defined, or literally `undefined`. If I add `q: textSearch` value to the object in the constructor, and it's undefined, an empty `?q=` gets output in the searchparams

File: test/functional/acl.test.js [hunk 2]
```diff
@@ -21,56 +21,74 @@ describe('HTTP API access control', () => {
...
+      'POST /api/registryusers/:id/unblock',
+      'GET /api/config',
+    ])
+
+    function* getRoutes(router, prefix = '') {
+      for (const item of router.stack) {
+        if (item.name === 'router') {
+          const prefix = item.regexp
+            .toString()
+            .match(/^\/\^\\(\/.*)\\\/\?\(\?=\\\/\|\$\)\/i$/i)[1]
```
[the remark this answers is not in the document]
Comment (googol) [reply test/functional/acl.test.js:44]: yeah that's a good point, and the regex could also be assigned to a constant for clarity

**Comment by googol** [comment]
Yeah I see this might seem a bit odd. The removal of the boot directory is actually probably the last step in a long chain of changes that work to remove the bootstrap framework from this project. Reki was originally started with that in use, and it turned out to be quite a bad decision for the needs that we had for the backend, so we wanted to get rid of it. The way the boot directory was laid out was based on the bootstrap style, even though the main framework has already been removed. I wanted to get rid of the directory since it isn't really normal style in an express app, but I haven't yet decided on a better style for factoring the routes and handlers. So I left that for a future PR

File: test/functional/acl.test.js [hunk 3]
```diff
@@ -21,56 +21,74 @@ describe('HTTP API access control', () => {
...
-        .filter((item) => !!item.route && !!item.route.path)
-        .flatMap((item) =>
-          _.map(
-            _.keys(item.route.methods),
-            (method) => `${method.toUpperCase()} ${item.route.path}`,
-          ),
-        )
-        .filter((item) => _.includes(item, ' /api')) //API endpoints only
-        .value()
+  describe.only('Access control tests', () => {
```
Comment (supertassu) [inline test/functional/acl.test.js:24]: Remove `.only`

**Comment by googol** [comment]
Yes, typescript conversion of the backend is next on my todo list, although I might do it in a couple of chunks.

Also, that separation of files you mentioned is quite close to what I like as well, except I prefer structuring the directories based on features rather than technical features to colocate all the code for a single feature (so more like `./places/controller.ts` and `./places/model.ts`), and wrapping up things like routers and route handlers in a function to enable explicit parameter passing

**Pull request merged**